src/config.rs

```diff
@@ -0,0 +1,2402 @@
+    /// "sina.com.cn" and "www.sina.com" but not "sina.cn".
+    Substring(String),
+    /// Predefined list of domain names: from the `"geosite:"` beginning, with
+    /// the remainder being a name, such as `geosite:google` or `geosite:cn` .
+    /// The domain name and reference list
+    /// [of predefined list of domain names](https://www.v2fly.org/config/routing.html#预定义域名列表) .
+    Predefined(String),
+    /// Loaded from a file domain: the form `"ext:file:tag"`, must begin with `ext:`(lowercase),
+    /// followed by the file name and label files are stored in the resource directory ,
+    /// the file format is `geosite.dat` the same, the label must exist in the file.
+    File(String),
+}
+
+pub enum NetworkType {
+    /// `"tcp"`
+    Tcp,
+    /// `"udp"`
+    Udp,
+    /// `"tcp,udp"`
+    TcpAndUdp,
+}
+
+/// The configuration file format of V2Ray is as follows, the client and server
+/// share the same format, but the actual configuration is different.
+///
+/// If you are new to V2Ray, you should start with understanding Inbounds and
+/// Outbounds, and start the program by filling in only the necessary options.
+/// Next, learn about other components step by step, you will find that V2Ray
+/// is not difficult to master.
+///
+/// ```json
+/// {
+///     "log": {},
+///     "api": {},
+///     "dns": {},
+///     "routing": {},
+///     "policy": {},
+///     "inbounds": [],
+///     "outbounds": [],
+///     "transport": {},
+///     "stats": {},
+///     "reverse": {}
+/// }
+/// ```
+pub struct Config {
+    /// Log configuration indicates how V2Ray outputs logs.
+    log: log::Log,
+    /// remote control.
+    api: api::Api,
+    /// The built-in DNS server, if this item does not exist, the DNS settings
+    /// of this machine will be used by default.
+    dns: dns::Dns,
+    /// Routing function.
+    routing: routing::Routing,
+    /// Local policies can be configured for some permissions.
+    policy: policy::Policy,
+    /// An array, each element is an inbound connection configuration.
+    inbounds: Vec<inbound::Inbound>,
+    /// An array, each element is an outbound connection configuration.
+    /// The first element in the list is the primary outbound protocol.
+    /// When the route matching does not exist or the matching is not
+    /// successful, the traffic is sent by the main outbound protocol.
+    outbounds: Vec<outbound::Outbound>,
+    /// Used to configure how V2Ray establishes and uses network connections
+    /// with other servers.
+    transport: transport::Transport,
+    /// Statistics.
+    stats: stats::Stats,
+    /// Reverse proxy.
+    reverse: reverse::Reverse,
+}
+
+pub mod log {
+    use serde::Deserialize;
+    /// The level of the log.
+    #[derive(Deserialize, Debug)]
+    pub enum LogLevel {
+        /// `"debug"` : Information that only developers can understand.
+        /// It contains all the `"info"` content.
+        #[serde(rename(deserialize = "debug"))]
```

Comment:
No need to rename them one-by-one as Serde will do the conversion between naming conventions.